hvantk/algorithms/qtlcascade/finemap.py

```
@@ -0,0 +1,294 @@
+                             note="fine-map skipped; missing: " + ", ".join(missing))
+    cache = _cache_dir(ld_cache_dir)
+    import tempfile
+    work = Path(work_dir) if work_dir else Path(tempfile.mkdtemp(prefix="hvantk_finemap_"))
+    work.mkdir(parents=True, exist_ok=True)
+    try:
+        n = _build_inputs(gwas, eqtl_recs, chrom, start, end, gwas_N, eqtl_N,
+                          cache, work, superpop)
+        if n < 2:
+            return FineMapResult(available=True, n_variants=n,
+                                 note="too few overlapping variants for fine-mapping")
+        p = subprocess.run(["Rscript", str(_R_SCRIPT), str(work)],
```

Ruff (0.15.15)

[error] `subprocess` call: check for execution of untrusted input

(S603)